### DISCUSSION TITLE
Error "bootstrap" MPM Application

### DISCUSSION THREAD

**Discussion by MiguelARV16** [KratosMultiphysics/Kratos#12738] [Q&A] [answered]
Hola soy Miguel Ángel.
Estoy trabajando con la aplicación de MPM en Kratos mediante GiD. Hace poco actualice la versión de Kratos y GiD y ahora al intentar realizar el cálculo me arroja el siguiente error:

Traceback (most recent call last):
  File "C:\mpm_ejemplo_flow_granular2D.gid\MainKratos.py", line 40, in <module>
    analysis_stage_module = importlib.import_module(analysis_stage_module_name)
  File "C:\Program Files\GiD\GiD 16.1.10d\scripts\tohil\python\lib\importlib\__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1050, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1027, in _find_and_load
  File "<frozen importlib._bootstrap>", line 992, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "<frozen importlib._bootstrap>", line 1050, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1027, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1004, in _find_and_load_unlocked
ModuleNotFoundError: No module named 'KratosMultiphysics.ParticleMechanicsApplication'

Por favor, ¿alguien podría orientarme?

Saludos, 
Miguel Ángel.

**Comment by rubenzorrilla** [comment, marked as answer]
Hola Miguel Ángel, es muy probablemente debido a este cambio #12011. En principio, debería resolverse cambiando el import de la `ParticleMechanicsApplication` a `MPMApplication`.

@ncrescenzio such application name changes are not quite common, but if it happens to you again in the future, it is a good practice to leave a "fake" application with the old name only implementing the init so to throw a warning when trying to import the corresponding module.

**Reply by ncrescenzio** [reply]
You are absolutely right. I am quite sure that when I changed the name to the application I did what you say, i.e. to leave the old `ParticleMechanicsApplication` with the aim of throwing a message, but for some reasons I probably forgot to add it to the pull request... my mistake. Should I add it now even if it's "late"?

**Reply by rubenzorrilla** [reply]
At this point I'd leave it as it is. It's been a while since you did the changes and there haven't been complaints so far.

**Comment by ncrescenzio** [comment]
@MiguelARV16 if GiD is connected to the `GiDInterface` from github, then you should have no problem, because everything is updated there as well. On the other hand, if you are not using the latest version then it should be sufficient to change `ParticleMechanicsApplication` with `MPMApplication` in the python script `MainKratos.py` and in the `ProjectParameters.json` file

**Reply by MiguelARV16** [reply]
I appreciate the prompt response. I tested with other applications and only got the error with the MPM App.

I made the connection through GiDInterface from github and it corrected part of the problem. Now it only indicates the following when calculating:

Traceback (most recent call last):
  File "C:\test_MPM_granular_flow.gid\MainKratos.py", line 41, in <module>
    analysis_stage_class = getattr(analysis_stage_module, analysis_stage_class_name)
                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: module 'KratosMultiphysics.MPMApplication.mpm_analysis' has no attribute 'MpmAnalysis'. Did you mean: 'MPMAnalysis'?

**Reply by ncrescenzio** [reply]
I think you are experiencing an issue that has been solved with #12562 and this means that the kratos version you are using is older than that pull request.

If you have installed Kratos by cloning the github repo and compiling it by yourself, then update the local version of the repo with `git pull` and compile kratos again.

Otherwise I think that it should be sufficient to open the file `MainKratos.py` generated by `GID` and replace the lines
```
analysis_stage_class_name = analysis_stage_module_name.split('.')[-1]
analysis_stage_class_name = ''.join(x.title() for x in analysis_stage_class_name.split('_'))
```
with the line
```
analysis_stage_class_name = "MPMAnalysis"
```

Let me know.

**Reply by MiguelARV16** [reply]
Thank you very much. The error has been resolved
